### PR TITLE
feat: add mobile header items and renderNavigation method

### PR DESCRIPTION
**Pull request by JeikZim**
### Feat 1 

You can add items to the header for mobile devices. Implemented using the NavigationList and OverflowScroller components.

**IPhone XR**
feat 1 on IPhone XR 1
feat 1 on IPhone XR 2
feat 1 on IPhone XR 3

https://github.com/user-attachments/assets/539e65fb-f7cd-496f-89c7-cdbe66960ee1

**IPad Mini**
feat 1 on IPad Mini

```
const customMobileHeaderItems = [{type: 'item1'}, {type: 'item2'}, {type: 'item3'}];

<PageConstructor
    custom={{
        navigation: {
            item3: () => ...,
        },
    }
    ...
    navigation={
         header: {
    ...
                 // new
                 mobileHeaderItems,
    ...
             }
        }
/>
```

### Feat 2 

In PageConstructor-component, you can pass the renderNavigation property to the navigation prop. This is a method that renders a custom navigation instead of default navigation (instead of Navigation-component).

preview of custom navigation

```
<PageConstructor
    ...
    navigation={
         renderNavigation: () => *some React.ReactNode*
     }
     ...
/>
```

### Feat 3 

Expanded the export from ./src/navigation. This allows you to reuse PC components (such as DesktopNavigation, MobileNavigation, NavigationList, etc.) when creating custom navigation in the "renderNavigation" method.

### Feat 4 

Refactor the Control component using the Button component from gravity-ui/uikit.

### CODE REVIEW

File: src/navigation/models.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export enum ItemColumnName {
 export enum NavigationLayout {
     Desktop = 'desktop',
     Mobile = 'mobile',
+    MobileHeader = 'mobile-header',
```
Comment (martyanovandrey) [inline src/navigation/models.ts:26]: better use css for this ?

File: src/navigation/components/MobileMenuButton/MobileMenuButton.tsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ export const MobileMenuButton: React.FC<MobileMenuButtonProps> = ({
 
             onSidebarOpenedChange(!isSidebarOpened);
         }}
-        size="l"
+        size={'xl'}
```
Comment (martyanovandrey) [inline src/navigation/components/MobileMenuButton/MobileMenuButton.tsx:27]: better not to change this size

File: src/models/navigation.ts [hunk 3]
```diff
@@ -90,6 +90,7 @@ export type ThemedNavigationLogoData = NavigationLogoData & ThemeSupporting<Navi
 export interface HeaderData {
     leftItems: NavigationItemModel[];
     rightItems?: NavigationItemModel[];
+    mobileHeaderItems?: NavigationItemModel[];
```
Comment (martyanovandrey) [inline src/models/navigation.ts:93]: consider using separate field in left/right Items

Comment (JeikZim) [reply src/models/navigation.ts]: We decided to keep this logic, but rename the name of the prop

File: src/navigation/components/MobileMenuButton/MobileMenuButton.tsx [hunk 4]
```diff
@@ -24,7 +24,7 @@ export const MobileMenuButton: React.FC<MobileMenuButtonProps> = ({
 
             onSidebarOpenedChange(!isSidebarOpened);
         }}
-        size="l"
+        size={'l'}
```
Comment (NikitaCG) [inline src/navigation/components/MobileMenuButton/MobileMenuButton.tsx:27]: I think this is an unnecessary change

File: src/navigation/components/Navigation/Navigation.tsx [hunk 5]
```diff
@@ -14,22 +19,17 @@ import './Navigation.scss';
...
-export const Navigation: React.FC<NavigationProps> = ({data, logo, className}) => {
-    const {
-        leftItems,
-        rightItems,
-        iconSize = 20,
-        withBorder = false,
-        withBorderOnScroll = true,
-    } = data;
-    const [isSidebarOpened, setIsSidebarOpened] = useState(false);
+export const useActiveNavItem = (
```
Comment (NikitaCG) [inline src/navigation/components/Navigation/Navigation.tsx:27]: move the custom hook to a separate file

File: src/navigation/components/Navigation/Navigation.tsx [hunk 6]
```diff
@@ -45,8 +45,11 @@ export const Navigation: React.FC<NavigationProps> = ({data, logo, className}) =
     const onActiveItemChange = (id?: string) => {
         setActiveItemId(id);
     };
+    return {activeItemId, leftItemsWithIconSize, rightItemsWithIconSize, onActiveItemChange};
+};
 
-    const onSidebarOpenedChange = (isOpen: boolean) => setIsSidebarOpened(isOpen);
+export const useShowBorder = (withBorder: boolean, withBorderOnScroll: boolean) => {
```
Comment (NikitaCG) [inline src/navigation/components/Navigation/Navigation.tsx:51]: and this too

**Review by NikitaCG — DISMISSED** [review]

**Review by NikitaCG — CHANGES_REQUESTED** [review]
see in the comments

**Review by NikitaCG — DISMISSED** [review]

**Comment by stepastepastepa** [comment]
About «Feat 1». Let's keep only one item in the preview on the right-hand side of the storybook https://preview.gravity-ui.com/page-constructor/989/?path=/story/navigation-navigation--navigation-with-custom-mobile-header-items, so that it's clear that the content differs from the desktop version.

**Review by NikitaCG — APPROVED** [review]
